### DISCUSSION TITLE
Is it required to use server folder in Nuxt 3?

### DISCUSSION THREAD

**Discussion by mojjjtaba** [nuxt/nuxt#27339] [Questions]
Hi everyone,
I want to build a full-stack application using Nuxt 3 and Supabase. we have a server folder in Nuxt3 should I use that? I'm using the Nuxt 3 Supabase module to communicate with the Supabase database. I can fetch or add data to tables in three was:
1. in each component
2. in composable
3. in server folder

which approach is best to be implemented? why should I use the server folder? why not?

**Comment by AloisSeckar** [comment]
From my point of view, if you don't need any special processing and you just handling basic CRUD operations, then using `useSupabaseClient` directly in components (or in dedicated composable if the logic repeats in more than one place) is perfectly fine. Because KISS. Nuxt server would essentialy become just a useless middle man adding complexity and obsfucation to the data flow.

Using server would start giving sense if you have for example some server middleware for auditing all requests or when you need to treat outgoing requests somehow before sending them to an external network (add some headers, alter URL, or whatever).

**Reply by mojjjtaba** [reply]
@AloisSeckar Thank you. I want to build an open-source project management system like Trello with Nuxt 3 and Supabase. I want to use it as a portfolio. what do you think? can you give me advice? with all of these things that I said you believe that the server folder is a time waster?

**Reply by AloisSeckar** [reply]
I am sorry, I can't forsee all possible requirements from such brief description. It depends, I guess. I would recommend you start with client-side Supabase. It shouldn't be that hard to extract the related logic and move it to server-side once you'll find it useful. I believe the biggest difference is that you need to use `useSupabaseServerClient`, but the underlying API remains the same.

**Comment by MustafaM257** [comment]
Hello,

First of all, choosing between these approaches depends heavily on the scalability and specific needs of each project. Personally, I prefer using the server directory.

This approach helps keep the code synchronized and takes advantage of the flexibility that Nuxt offers. You can easily manage the server-only parts of your application, from API endpoints to middleware.

If you’re working on a project management application, scalability will hit, and you'll likely encounter various obstacles. Pushing for  the server directory allows you to structure your endpoints in a way that supports scaling and efficient management.

So, that's my perspective. Use the server directory to its fullest potential to ensure your application can grow and adapt to future challenges.

Read more about server dir as well, you'll have a better view and approaches, it might not be what you need!

**Reply by mojjjtaba** [reply]
Thank you Mustafa
